pyarlo/base_station.py

```diff
@@ -239,8 +239,9 @@ def mode(self):
             modes = mode_event['properties']['modes']
             for mode in modes:
                 if mode['id'] == active_mode:
-                    return mode['type']
-
+                    if 'type' in mode:
+                        return mode['type']
+                    return mode.get('name', None)
```

Comment:
@chaddotson let's change the logic here a little bit. How about this:
```python
.....
if mode['id'] == active_mode:
  return mode.get('type') if mode.get('type') is not None else mode.get('name')
```
By default mode.get() will return None if 'name' does not exist, so you don't need the "None" there.